src/localization/gps_linearization.py

```diff
@@ -37,20 +41,33 @@ class GPSLinearization:
     def __init__(self):
         # read required parameters, if they don't exist an error will be thrown
         self.ref_lat = rospy.get_param("gps_linearization/reference_point_latitude")
-        self.ref_lon = rospy.get_param("gps_linearization/reference_point_longitude")
+        self.ref_lon = rospy.get_param(param_name="gps_linearization/reference_point_longitude")
         self.ref_alt = rospy.get_param("gps_linearization/reference_point_altitude")
+        self.both_gps = rospy.get_param("gps_linearization/use_both_gps")
         self.world_frame = rospy.get_param("world_frame")
         self.use_dop_covariance = rospy.get_param("global_ekf/use_gps_dop_covariance")
+        self.ref_coord = np.array([self.ref_lat, self.ref_lon, self.ref_alt])
+
+        # config gps and imu convariance matrices
         self.config_gps_covariance = np.array(rospy.get_param("global_ekf/gps_covariance", None))
+        self.config_imu_covariance = np.array(rospy.get_param("global_ekf/imu_orientation_covariance", None))
 
         self.last_gps_msg = None
+        self.last_pose = None
         self.last_imu_msg = None
 
-        rospy.Subscriber("gps/fix", NavSatFix, self.gps_callback)
+        if self.both_gps:
+            right_gps_sub = message_filters.Subscriber("right_gps_driver/fix", NavSatFix)
+            left_gps_sub = message_filters.Subscriber("left_gps_driver/fix", NavSatFix)
+            sync_gps_sub = message_filters.ApproximateTimeSynchronizer([right_gps_sub, left_gps_sub], 10, 0.5)
```

Comment:
I would make 10 and 0.5 constant variables at the top of hte file

Reply:
fixed